chainer/variable.py

```diff
@@ -1273,18 +1274,21 @@ def __reduce__(self):
     def to_cpu(self):
         super(Parameter, self).to_cpu()
         if self.data is None:
+            self._initial_backend = None
             self._initial_device = None
 
     def to_gpu(self, device=None):
         super(Parameter, self).to_gpu(device)
         if self.data is None:
             if device is None:
                 device = cuda.Device().id
+            self._initial_backend = 'cuda'
             self._initial_device = device
 
     def to_intel64(self):
         super(Parameter, self).to_intel64()
         if self.data is None:
+            self._initial_backend = 'intel64'
```

Comment:
I think strings are in general not optimal for this kind of use, unless it's interned.
Maybe it does not matter because `to_xxx` methods are inherently not meant to be fast.

Do you have any thoughts?

Reply:
I interned these string constants.
When more backends are added, maybe it's better to define them in each backend like `chainer.backneds.cuda.ID`.

Reply:
What I meant was to use `sys.intern` and `==` for comparison (maybe overkill).
In current implementation, there's no reason to use strings.
How about either using unique objects (i.e. `object()`) or going back to simple string comparison with `==`?

Reply:
Ah, I see. I reverted back to the previous code anyway.